src/utils/WeakMap.js

```diff
@@ -0,0 +1,9 @@
+class NonTrackingDummyWeakMap {
+  get() {}
+  set() {}
```

Comment:
Might be worth retuning `this` so that our `NonTrackingDummyWeakMap` behaves like `WeakMap` with regard to chaining.

Reply:
@matthieuprat ok updated